test/functional/core/startup_spec.lua

```diff
@@ -494,6 +495,60 @@ describe('user config init', function()
   end)
 end)
 
+describe('runtime:', function()
+  local xhome = 'Xhome'
+  local pathsep = helpers.get_pathsep()
+  local xconfig = xhome .. pathsep .. 'Xconfig'
+
+  setup(function()
+    mkdir_p(xconfig .. pathsep .. 'nvim')
+  end)
+
+  teardown(function()
+    rmdir(xhome)
+  end)
+
+  it('loads plugin/*.lua from XDG config home', function()
+    local plugin_folder_path = table.concat({xconfig, 'nvim', 'plugin'}, pathsep)
+    local plugin_file_path = table.concat({plugin_folder_path, 'plugin.lua'}, pathsep)
+    mkdir_p(plugin_folder_path)
+    write_file(plugin_file_path, [[ vim.g.lua_plugin = 1 ]])
+
+    clear{ args_rm={'-u'}, env={ XDG_CONFIG_HOME=xconfig }}
+
+    eq(1, eval('g:lua_plugin'))
+    rmdir(plugin_folder_path)
+  end)
+
+  it('loads plugin/*.lua from start plugins', function()
+    local plugin_path = table.concat({xconfig, 'nvim', 'pack', 'catagory',
+    'start', 'test_plugin'}, pathsep)
+    local plugin_folder_path = table.concat({plugin_path, 'plugin'}, pathsep)
+    local plugin_file_path = table.concat({plugin_folder_path, 'plugin.lua'},
+    pathsep)
+    mkdir_p(plugin_folder_path)
+    write_file(plugin_file_path, [[vim.g.lua_plugin = 2]])
+
+    clear{ args_rm={'-u'}, env={ XDG_CONFIG_HOME=xconfig }}
+
+    eq(2, eval('g:lua_plugin'))
+    rmdir(plugin_path)
+  end)
+
+  it('loads ftdetect/*.lua', function()
+    local ftdetect_folder = table.concat({xconfig, 'nvim', 'ftdetect'}, pathsep)
+    local ftdetect_file = table.concat({ftdetect_folder , 'new-ft.lua'}, pathsep)
+    mkdir_p(ftdetect_folder)
+    write_file(ftdetect_file , [[vim.g.lua_ftdetect = 1]])
+
+    clear{ args_rm={'-u'}, env={XDG_CONFIG_HOME=xconfig,
+                                VIMRUNTIME='runtime/'}}
```

Comment:
@bfredl any idea why this test fail if I don't set `VIMRUNTIME` env ?

Reply:
I'd be fine with adding a TODO comment for now

Reply:
Ok . I've added a todo . Will look into why it happens later